lib/vanagon/platform/windows.rb

```diff
@@ -289,6 +289,105 @@ def output_dir(target_repo = "")
         File.join("windows", target_repo, @architecture)
       end
 
+      # Generate correctly formatted wix elements that match the
+      # structure of the directory input
+      #
+      # @param services, Array of components services
+      # and optionally:
+      # @return [string] correctly formatted wix element string
+      def generate_wix_dirs(services)
+        directories = []
+        services.map { |svc| directories.push({ :path => svc.service_file, :id => "#{svc.id}BINDIR" }) }
```

Comment:
Yes, So what it does is strip off SourceDir, because SourceDir corresponds to TARGETDIR with is the directoryRef already existing in directorylist.wxs.erb, then it will strip off the last entry in the path (in this case nssm.exe) because we want it's source directory. Then you'll notice that https://github.com/McdonaldSeanp/vanagon/blob/feature/RE-6455/dirlist/lib/vanagon/platform/windows.rb#L379L381 puts several child elements (without "name" attributes) under the actual bin directory. Per the WiX documentation, if you add a directory element without a name under another directory element with a name, both elements will reference the same directory with different Ids, which is what we want.